scripts/ci/run-test-compiler-server.sh

```diff
@@ -0,0 +1,14 @@
+#!/bin/bash -e
```

Comment:
Do we need extra test step? I though we will enable it everywhere for normal build

Reply:
It's not enabled everywhere right now, and this also verifies that it works so it makes sense for it to be an extra test step

The compiler server is designed to silently fail in like a dozen scenarios so we'll always need a dedicated test for it